apps/bsd/src/screens/WriteInAdjudicationScreen.tsx

```diff
@@ -0,0 +1,473 @@
+  electionDefinition,
+  imageURL,
+  interpretation,
+  layout,
+  contestIds,
+  writeInValues,
+  onContestChange,
+  onWriteInChanged,
+  onAdjudicationComplete,
+}: {
+  electionDefinition: ElectionDefinition
+  imageURL: string
+  interpretation: InterpretedHmpbPage
+  layout: SerializableBallotPageLayout
+  contestIds: readonly Contest['id'][]
+  writeInValues: WriteInsByContestAndOption
+  onContestChange?(contestId?: Contest['id']): void
+  onWriteInChanged?(
+    contestId: Contest['id'],
+    optionId: ContestOption['id'],
+    value: string
+  ): void
+  onAdjudicationComplete(): Promise<void>
+}): JSX.Element => {
+  const makeCancelable = useCancelablePromise()
+  const [isSaving, setIsSaving] = useState(false)
+  const [selectedContestIndex, setSelectedContestIndex] = useState(0)
+
+  const writeIns = interpretation.adjudicationInfo.allReasonInfos.filter(
+    (reason): reason is WriteInAdjudicationReasonInfo =>
+      reason.type === AdjudicationReason.WriteIn
+  )
+  const contestsWithWriteIns = new Set(
+    [...writeIns].map(({ contestId }) => contestId)
+  )
+  const contestsWithWriteInsCount = contestsWithWriteIns.size
+  const selectedContestId = [...contestsWithWriteIns][selectedContestIndex]
+  const isFirstContestSelected = selectedContestIndex === 0
+  const isLastContestSelected =
+    selectedContestIndex === contestsWithWriteInsCount - 1
+
+  const onWriteInInput = useCallback(
+    (optionId: ContestOption['id'], value: string) => {
+      onWriteInChanged?.(selectedContestId, optionId, value)
+    },
+    [onWriteInChanged, selectedContestId]
+  )
+
+  useEffect(() => {
+    onContestChange?.(selectedContestId)
+  }, [selectedContestId, onContestChange])
+
+  const contestIndex = contestIds.findIndex((id) => id === selectedContestId)
+  const contestsWithWriteInsIndex = [...contestsWithWriteIns].findIndex(
+    (id) => id === selectedContestId
+  )
+  const contest = find(
+    electionDefinition.election.contests,
+    (c): c is CandidateContest => c.id === selectedContestId
+  )
+  const writeInsForContest = writeIns.filter(
+    (writeIn) => writeIn.contestId === selectedContestId
+  )
+  const contestLayout = layout.contests[contestIndex]
+  const allWriteInsHaveValues = writeInsForContest.every(
+    (writeIn) => writeInValues[selectedContestId]?.[writeIn.optionId]
+  )
+
+  const goPrevious = useCallback(() => {
+    setSelectedContestIndex((prev) => Math.max(0, prev - 1))
+  }, [])
+  const goNext = useCallback(
+    async (event?: React.FormEvent<EventTarget>): Promise<void> => {
+      event?.preventDefault()
+      if (isLastContestSelected) {
+        setIsSaving(true)
+        try {
+          await makeCancelable(onAdjudicationComplete())
+        } finally {
+          setIsSaving(false)
+        }
+      } else {
+        setSelectedContestIndex((prev) => prev + 1)
+      }
+    },
+    [isLastContestSelected, makeCancelable, onAdjudicationComplete]
+  )
+
+  return (
+    <form onSubmit={goNext}>
```

Comment:
oh my goodness, an actual HTML element that is not a React element? I am shook.

Reply:
![image](https://user-images.githubusercontent.com/1938/133297046-0be810e8-ec0e-4994-b3d4-d4a0c08b0be4.png)